runtime/lua/vim/diagnostic.lua

```diff
@@ -346,18 +346,15 @@ local function schedule_display(namespace, bufnr, args)
   bufs_waiting_to_update[bufnr][namespace] = args
 
   local key = make_augroup_key(namespace, bufnr)
+  local group = vim.api.nvim_create_augroup(key)
   if not registered_autocmds[key] then
-    vim.cmd(string.format(
-      [[augroup %s
-      au!
-      autocmd %s <buffer=%s> lua vim.diagnostic._execute_scheduled_display(%s, %s)
-    augroup END]],
-      key,
-      table.concat(insert_leave_auto_cmds, ','),
-      bufnr,
-      namespace,
-      bufnr
-    ))
+    vim.api.nvim_create_autocmd(insert_leave_auto_cmds, {
+      group = group,
+      buffer = bufnr,
+      callback = function()
+        vim.diagnostic._execute_scheduled_display(namespace, bufnr)
```

Comment:
We should be able to make `_execute_scheduled_display` a local function now that native Lua callbacks are supported. Can do that in a follow up though